Lib/dataclasses.py

```diff
@@ -1257,16 +1260,17 @@ def _add_slots(cls, is_frozen, weakref_slot):
     )
     # The slots for our class.  Remove slots from our base classes.  Add
     # '__weakref__' if weakref_slot was given, unless it is already present.
-    cls_dict["__slots__"] = tuple(
-        itertools.filterfalse(
+    cls_dict["__slots__"] = {
```

Comment:
Idea: maybe we can only change the format to `dict` when there's at least one `doc` field?

In other cases we can continue to return `tuple`

Reply:
I was considering that, but the bookkeeping needed to make the decision seemed like a hassle. But I do think it would be a good idea, if the code isn't too bad.

Reply:
I will try this out! 👍